pkg/agent/openflow/client.go

```diff
@@ -898,6 +903,13 @@ func (c *client) generatePipelines() {
 		c.activatedFeatures = append(c.activatedFeatures, c.featureExternalNodeConnectivity)
 	}
 
+	grpAllocator := c.v4GroupIDAllocator
+	for _, proto := range c.ipProtocols {
+		if proto == binding.ProtocolIPv6 {
+			grpAllocator = c.v6GroupIDAllocator
+			break
```

Comment:
Not sure I follow this. So v6GroupIDAllocator is used for dual stack cases?

Reply:
Yes, we only need a single group allocator no matter it is IPv4 only, IPv6 only or dual-stack because the buckets in the groups are static. So here I choose IPv6 groupAllocator as long as IPv6 is enabled (which is to ensure the allocator is not nil).

Reply:
The v4 group allocator is not nil anyway regardless of IPv4, IPv6, or dual-stack.

Reply:
https://github.com/antrea-io/antrea/pull/5188 will merge the group allocators so we don't have to introduce weird code.

Reply:
updated to use IPv4 groupIDAllocator in all sceraios in the latest change. I will rebase after #5188 is merged.